log4j-api/src/main/java/org/apache/logging/log4j/util/PropertySource.java

```diff
@@ -137,9 +137,12 @@ final class Util {
          * @param value property name
          * @return the property broken into lower case tokens
          */
+        // https://errorprone.info/bugpattern/CollectionUndefinedEquality
+        @SuppressWarnings("CollectionUndefinedEquality")
         public static List<CharSequence> tokenize(final CharSequence value) {
-            if (CACHE.containsKey(value)) {
-                return CACHE.get(value);
+            // `value` should be a `String`
+            if (CACHE.containsKey(value.toString())) {
+                return CACHE.get(value.toString());
```

Comment:
Oops, I think this was my bad.

Reply:
I noticed this before, but the method is only called with `String` parameters, so it is not actually a problem.